### PR TITLE
Reducing time of running tests

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/parallel/council-pallet.API.foundation.maintenance.test.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      councilUsers[i],
+      Council.vote(
+        hash,
+        JSON.parse(JSON.stringify(proposal.toHuman())).index,
+        "aye",
+      ),
+    );
+    txs.push(tx);
+  }
+  return await Sudo.batchAsSudoFinalized(...txs);
```
Comment (goncer) [inline test/parallel/council-pallet.API.foundation.maintenance.test.ts:244]: I guess that those aux functions are duplicated on both tests,  could you please reuse them?

File: test/parallel/xyk-pallet.API.parachainStacking.joinCandidate.test.ts [hunk 2]
```diff
@@ -78,34 +85,34 @@ beforeAll(async () => {
   await Sudo.asSudoFinalized(
     Sudo.sudo(Staking.addStakingLiquidityToken(liqToken)),
   );
-});
 
-beforeEach(async () => {
-  testUser1 = new User(keyring);
+  [testUser1, testUser2, testUser3, testUser4] = setupUsers();
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.parachainStacking.joinCandidate.test.ts:89]: why this? and consequent testUser changes?

File: test/sequential/bootstrap.updatePromotion.test.ts [hunk 3]
```diff
@@ -95,7 +90,12 @@ test("bootstrap - bootstrap - Check if we can change promoteBootstrapPool in eac
   );
   await waitSudoOperationSuccess(scheduleBootstrapBefPlan);
 
-  checkingUpdatingPool = await changePromotionBootstrapPool(sudo);
+  const startingPromotionState = await getPromotionBootstrapPoolState();
+
+  checkingUpdatingPool = await updatePromoteBootstrapPool(
```
Comment (goncer) [inline test/sequential/bootstrap.updatePromotion.test.ts:95]: plz rename

File: test/sequential/bootstrap.updatePromotion.test.ts [hunk 4]
```diff
@@ -82,7 +77,7 @@ beforeAll(async () => {
   await setupBootstrapTokensBalance(bootstrapCurrency, sudo, [testUser1]);
 });
 
-test("bootstrap - bootstrap - Check if we can change promoteBootstrapPool in each phase", async () => {
+test("bootstrap - Check if we can change promoteBootstrapPool in each phase", async () => {
   let checkingUpdatingPool: MangataGenericEvent[];
```
Comment (goncer) [inline test/sequential/bootstrap.updatePromotion.test.ts:81]: plz rename , and: why are you using this global variable ?

File: test/sequential/xyk-pallet.asset.bootstrap.register.test.ts [hunk 5]
```diff
@@ -74,13 +70,12 @@ async function runBootstrap(assetId: BN) {
...
-  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
-
-  const bootstrapFinalize = await finalizeBootstrap(sudo);
-  eventResponse = getEventResultFromMangataTx(bootstrapFinalize);
-  expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  const claimRewardsAndBootstrapFinalize = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, api.tx.bootstrap.claimLiquidityTokens()),
+    Sudo.sudoAs(sudo, api.tx.bootstrap.preFinalize()),
+    Sudo.sudoAs(sudo, api.tx.bootstrap.finalize()),
+  );
```
Comment (goncer) [inline test/sequential/xyk-pallet.asset.bootstrap.register.test.ts:77]: :clap:

File: utils/Bootstrap.ts [hunk 6]
```diff
@@ -165,27 +164,9 @@ export async function claimAndActivateBootstrap(user: User) {
...
-      api.tx.bootstrap.finalize(),
-    ),
-    sudoUser.keyRingPair,
-    {
-      nonce: await getCurrentNonce(sudoUser.keyRingPair.address),
-    },
+  return await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(sudoUser, api.tx.bootstrap.preFinalize()),
+    Sudo.sudoAs(sudoUser, api.tx.bootstrap.finalize()),
   );
```
Comment (goncer) [inline utils/Bootstrap.ts:170]: :clap:

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/parallel/proofOfStake.3rdPartyRewards.calim.activated.test.ts [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+          MGA_ASSET_ID,
+          Assets.DEFAULT_AMOUNT.muln(20e6),
+          newToken3,
+          Assets.DEFAULT_AMOUNT.muln(20e6),
+        ),
+      ),
+    );
+    liqIdMgaToken1 = await getLiquidityAssetId(MGA_ASSET_ID, newToken1);
+    liqIdMgaToken2 = await getLiquidityAssetId(MGA_ASSET_ID, newToken2);
+    liqIdMgaToken3 = await getLiquidityAssetId(MGA_ASSET_ID, newToken3);
```
[the remark this answers is not in the document]
Comment (AlexChetverov) [reply test/parallel/proofOfStake.3rdPartyRewards.calim.activated.test.ts:88]: This need to promote pools and get rewards

File: test/parallel/council-pallet.API.foundation.maintenance.test.ts (hunk 1 above)
Comment (AlexChetverov) [reply test/parallel/council-pallet.API.foundation.maintenance.test.ts:244]: Done

File: test/sequential/bootstrap.updatePromotion.test.ts (hunk 3 above)
Comment (AlexChetverov) [reply test/sequential/bootstrap.updatePromotion.test.ts:95]: done

File: test/sequential/bootstrap.updatePromotion.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply test/sequential/bootstrap.updatePromotion.test.ts]: I need to use this action several times per test.

**Review by goncer — APPROVED** [review]

**Pull request merged**